apm-agent-plugins/apm-jaxrs-httpclient-plugin/src/test/java/co/elastic/apm/agent/jaxrs/httpclient/JaxRsHttpClientInstrumentationTest.java

```diff
@@ -0,0 +1,51 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 - 2019 Elastic and contributors
+ * %%
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.jaxrs.httpclient;
+
+import co.elastic.apm.agent.httpclient.AbstractHttpClientInstrumentationTest;
+import org.junit.Before;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+public class JaxRsHttpClientInstrumentationTest extends AbstractHttpClientInstrumentationTest {
+
+    private Client client;
+
+    @Before
+    public void setUp() {
+        this.client = ClientBuilder.newClient();
+    }
+
+    @Override
+    protected void performGet(String path) throws Exception {
+        Response response = client
+            .target(path)
+            .request(MediaType.APPLICATION_JSON)
```

Comment:
```suggestion
            .request(MediaType.APPLICATION_JSON)
            .async()
            .get()
```

Does this pass as well?

Reply:
No, it doesn't.
```
main] INFO wiremock.org.eclipse.jetty.server.handler.ContextHandler - Started w.o.e.j.s.ServletContextHandler@75aea2ba{/__admin,null,AVAILABLE}
[main] INFO wiremock.org.eclipse.jetty.server.handler.ContextHandler - Started w.o.e.j.s.ServletContextHandler@5ff6dd3c{/,null,AVAILABLE}
[main] INFO wiremock.org.eclipse.jetty.server.NetworkTrafficServerConnector - Started NetworkTrafficServerConnector@5fcd1f0a{HTTP/1.1}{0.0.0.0:34203}
[main] INFO wiremock.org.eclipse.jetty.server.Server - Started @6918ms
[main] DEBUG co.elastic.apm.agent.impl.transaction.AbstractSpan - increment references to '' 00-4b9e8b4c67e4bf89081eddf7b1db6e34-9708b53297d4bf83-01 (536b71b4) (1)
[main] DEBUG co.elastic.apm.agent.impl.ElasticApmTracer - startTransaction '' 00-4b9e8b4c67e4bf89081eddf7b1db6e34-9708b53297d4bf83-01 (536b71b4) {
[main] DEBUG co.elastic.apm.agent.impl.transaction.AbstractSpan - increment references to '' 00-4b9e8b4c67e4bf89081eddf7b1db6e34-9708b53297d4bf83-01 (536b71b4) (2)
[main] DEBUG co.elastic.apm.agent.impl.ElasticApmTracer - Activating '' 00-4b9e8b4c67e4bf89081eddf7b1db6e34-9708b53297d4bf83-01 (536b71b4) on thread 1
[qtp137685382-62] INFO / - RequestHandlerClass from context returned com.github.tomakehurst.wiremock.http.StubRequestHandler. Normalized mapped under returned 'null'
[main] DEBUG co.elastic.apm.agent.impl.ElasticApmTracer - Deactivating '' 00-4b9e8b4c67e4bf89081eddf7b1db6e34-9708b53297d4bf83-01 (536b71b4) on thread 1
[main] DEBUG co.elastic.apm.agent.impl.transaction.AbstractSpan - decrement references to '' 00-4b9e8b4c67e4bf89081eddf7b1db6e34-9708b53297d4bf83-01 (536b71b4) (1)
[main] DEBUG co.elastic.apm.agent.impl.ElasticApmTracer - } endTransaction 'unnamed' 00-4b9e8b4c67e4bf89081eddf7b1db6e34-9708b53297d4bf83-01 (536b71b4)
[main] INFO wiremock.org.eclipse.jetty.server.NetworkTrafficServerConnector - Stopped NetworkTrafficServerConnector@5fcd1f0a{HTTP/1.1}{0.0.0.0:0}
[main] INFO wiremock.org.eclipse.jetty.server.handler.ContextHandler - Stopped w.o.e.j.s.ServletContextHandler@5ff6dd3c{/,null,UNAVAILABLE}
[main] INFO wiremock.org.eclipse.jetty.server.handler.ContextHandler - Stopped w.o.e.j.s.ServletContextHandler@75aea2ba{/__admin,null,UNAVAILABLE}
[main] WARN wiremock.org.eclipse.jetty.util.thread.QueuedThreadPool - qtp137685382{STOPPING,8<=8<=10,i=7,q=6} Couldn't stop Thread[qtp137685382-64,5,main]
[main] WARN wiremock.org.eclipse.jetty.util.thread.QueuedThreadPool - qtp137685382{STOPPING,8<=8<=10,i=3,q=6} Couldn't stop Thread[qtp137685382-59,5,]
[main] WARN wiremock.org.eclipse.jetty.util.thread.QueuedThreadPool - qtp137685382{STOPPING,8<=8<=10,i=0,q=5} Couldn't stop Thread[qtp137685382-68,5,]

java.lang.IndexOutOfBoundsException: Index 0 out-of-bounds for length 0

	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
```

Reply:
So, I guess this reproduces the problem originally reported in https://discuss.elastic.co/t/jaxrs-async-distrubuted-tracing-not-works/189531, right?